packages/react-router/src/path.ts

```diff
@@ -222,7 +222,10 @@ export function interpolatePath({
       // the splat/catch-all routes shouldn't have the '/' encoded out
       encodedParams[key] = isValueString ? encodeURI(value) : value
     } else {
-      encodedParams[key] = isValueString ? encodeURIComponent(value) : value
+      encodedParams[key] = isValueString
+        ? // allow '@' in path segments
+          encodeURIComponent(value).replaceAll('%40', '@')
```

Comment:
This wouldn't really scale well should additional characters need to be escaped.

Perhaps this `replaceAll` call should be moved into a function, where it can be referenced against an object by its keys and replaced with its values.

Maybe something like this?

```js
function unescapeParamValues(str) {
	const obj = { '%40': '@' }
	const enc = encodeURIComponent(str)
	Object.entries(obj).forEach(([key, val]) => {
		enc.replaceAll(key, val)
	})
	return enc
}
```

Reply:
Thanks @SeanCassiere for the feedback. See latest for changes.